### PR TITLE
Allow variant to be overridden from URL fragment

### PR DESCRIPTION
**Pull request by lannka**
#1411

A re-try to address comments in #4079

### CODE REVIEW

File: extensions/amp-experiment/0.1/variant.js [hunk 1]
```diff
@@ -34,6 +36,12 @@ export function allocateVariant(win, experimentName, config) {
   assertName(experimentName);
   validateConfig(config);
 
+  // Variant can be overridden from URL fragment.
+  const override = viewerFor(win).getParam(ATTR_PREFIX + experimentName);
+  if (override && config.variants.hasOwnProperty(override)) {
```
Comment (aghassemi) [inline extensions/amp-experiment/0.1/variant.js]: For stronger safety, I suggest doing this in `getVariantOverride_()` so invalid values never make it to the overrides map to start with.

Comment (lannka) [inline extensions/amp-experiment/0.1/variant.js]: Sorry, I don't quite understand. Are you suggesting to extract a method named `getVariantOverride_()`? Why would that be safer?

Comment (aghassemi) [inline extensions/amp-experiment/0.1/variant.js]: sorry this comment was based on the code in  https://github.com/ampproject/amphtml/pull/4079. I see that PR is closed. This is fine, ignore my previous comment.

File: extensions/amp-experiment/0.1/test/test-variant.js [hunk 2]
```diff
@@ -167,6 +177,28 @@ describe('allocateVariant', () => {
...
+    getParamStub.withArgs('amp-x-name').returns('-Variant_3');
+    return expect(allocateVariant(fakeWin, 'name', {
+      sticky: false,
+      variants: {
+        '-Variant_1': 50,
+        '-Variant_2': 50,
+      },
+    })).to.eventually.equal('-Variant_2');
+  });
+
```
Comment (aghassemi) [inline extensions/amp-experiment/0.1/test/test-variant.js]: nit: a test for case-sensitivity would be nice here to protect against `getParam` suddenly becoming case-insensitive and breaking this. (there are some web frameworks that lower-case QS params).

**Comment by aghassemi** [comment]
LGTM

File: extensions/amp-experiment/0.1/test/test-variant.js (hunk 2 above)
Comment (lannka) [inline extensions/amp-experiment/0.1/test/test-variant.js]: Sounds good. Done.

**Comment by lannka** [comment]
Comment addressed. PTAL

**Comment by jridgewell** [comment]
LGTM.

**Pull request merged**